main.go

```diff
@@ -6,8 +6,14 @@ import (
 	"fmt"
 	"os"
 
+	"cloud.google.com/go/profiler"
 	"github.com/spacemeshos/go-spacemesh/cmd"
 	"github.com/spacemeshos/go-spacemesh/cmd/node"
+	cfg "github.com/spacemeshos/go-spacemesh/config"
+)
+
+var (
+	config = cfg.DefaultConfig()
 )
```

Comment:
This is no good, it means that you'll always look at the **default** config, ignoring the **actual** config (e.g. config file, command line flags). For this reason, the call to `profiler.Start()` cannot happen in `main.go` and must be moved to after the config has been resolved.

I would put it in `cmd/node/node.go`, inside the `Initialize()` method, after the call to `cmdp.EnsureCLIFlags()`.